adcc/ExcitedStates.py

```diff
@@ -657,6 +659,66 @@ def describe_amplitudes(self, tolerance=0.01, index_format=None):
                 ret += separator
         return ret[:-1]
 
+    def to_dataframe(self):
+        """
+        Exports the ExcitedStates object as :class:`pandas.DataFrame`.
+        All values are printed in atomic units.
+        """
+        propkeys = self.excitation_property_keys.copy()
+        propkeys.extend(self.excitation_energy_corrections.keys())
+        data = {
+            "excitation": np.arange(0, self.size, dtype=int),
+            "kind": np.tile(self.kind, self.size)
+        }
+        # fill the dict with data, sort out stuff that we cannot represent
+        # in a table
+        unacceptable_types = [adcc.AmplitudeVector, adcc.OneParticleOperator,
+                              adcc.Tensor]
+        for key in propkeys:
+            try:
+                d = getattr(self, key)
+            except NotImplementedError:
+                # some properties are not available for every backend
+                continue
+            if isinstance(d, list):
+                if not any(type(e) in unacceptable_types for e in d):
+                    data[key] = d
+            elif isinstance(d, np.ndarray):
+                if d.ndim == 1:
+                    data[key] = d
+                elif d.ndim == 2 and d.shape[1] == 3:
+                    for i, p in enumerate(["x", "y", "z"]):
+                        data[f"{key}_{p}"] = d[:, i]
```

Comment:
This quite deeply nested. I don't have a good idea how to not do it, but maybe when you refactor you have one.

Reply:
Not as deeply nested in the new version 👍